shared/gcs/gcs.go

```diff
@@ -48,13 +48,13 @@ func Authenticate(ctx context.Context, serviceAccount string) error {
 func Exists(ctx context.Context, bucketName, storagePath string) bool {
 	// Check if this is a file
 	handle := createStorageObject(bucketName, storagePath)
-	if _, err := handle.Attrs(ctx); nil == err {
+	if _, err := handle.Attrs(ctx); err == nil {
```

Comment:
these can be combined
```
if _, err := handle.Attrs(); err != nil {
  _, err = getObjectsIter().Next()
  return err == nil
}
return true
```

Reply:
I would prefer current logic: return true if it's a file, else check if it's a folder.